openstack/baremetal/v1/ports/doc.go

```diff
@@ -0,0 +1,73 @@
+package ports
+
+/*
+ Package ports contains the functionality to Listing, Searching, Creating, Updating,
+ and Deleting of bare metal Port resources
+
+ API reference: https://developer.openstack.org/api-ref/baremetal/#ports-ports
+
+
+ 	// Example to List Ports with Detail
+ 	ports.ListDetail(client, ports.ListOpts{}).EachPage(func(page pagination.Page) (bool, error) {
+ 		portList, err := ports.ExtractPorts(page)
+ 		if err != nil {
+ 			return false, err
+ 		}
+
+ 		for _, n := range portList {
+ 			// Do something
+ 		}
+
+ 		return true, nil
+ 	})
+
+ 	// Example to List Ports
+ 	ports.List(client, ports.ListOpts{
+ 		Limit: 10,
+ 	}).EachPage(func(page pagination.Page) (bool, error) {
+ 		portList, err := ports.ExtractPorts(page)
+ 		if err != nil {
+ 			return false, err
+ 		}
+
+ 		for _, n := range portList {
+ 			// Do something
+ 		}
+
+ 		return true, nil
+ 	})
+
+ 	// Example to Create a Port
+ 	createPort, err := ports.Create(client, ports.CreateOpts{
+ 		NodeUUID: "e8920409-e07e-41bb-8cc1-72acb103e2dd",
+ 		Address: "00-1B-63-84-45-E6",
```

Comment:
I think ironic accepts addresses only separated by colon. I may be wrong though, needs double-checking.

Reply:
yeah I will update